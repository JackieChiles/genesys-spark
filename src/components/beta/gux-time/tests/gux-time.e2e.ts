import { newSparkE2EPage, a11yCheck } from '../../../../../tests/e2eTestUtils';

const date = new Date(2022, 6, 7, 9, 35, 30, 100);

describe('gux-time-beta', () => {
  describe('#render', () => {
    [
        `<gux-time-beta format="short"></gux-time-beta>`,
        `<gux-time-beta format="medium"></gux-time-beta>`,
        `<gux-time-beta format="long"></gux-time-beta>`,
        `<gux-time-beta format="full"></gux-time-beta>`
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-time-beta');

        element.setAttribute('date', date);
        await page.waitForChanges()

        await a11yCheck(page);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
