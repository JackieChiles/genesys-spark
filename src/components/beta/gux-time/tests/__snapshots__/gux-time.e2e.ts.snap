// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-time-beta #render should render component as expected (1) 1`] = `"<gux-time-beta format=\\"short\\" hydrated=\\"\\" date=\\"2022-07-07T13:35:30.100Z\\"></gux-time-beta>"`;

exports[`gux-time-beta #render should render component as expected (2) 1`] = `"<gux-time-beta format=\\"medium\\" hydrated=\\"\\" date=\\"2022-07-07T13:35:30.100Z\\"></gux-time-beta>"`;

exports[`gux-time-beta #render should render component as expected (3) 1`] = `"<gux-time-beta format=\\"long\\" hydrated=\\"\\" date=\\"2022-07-07T13:35:30.100Z\\"></gux-time-beta>"`;

exports[`gux-time-beta #render should render component as expected (4) 1`] = `"<gux-time-beta format=\\"full\\" hydrated=\\"\\" date=\\"2022-07-07T13:35:30.100Z\\"></gux-time-beta>"`;
